### PR TITLE
Setup Git Hooks and pre-commit hooks

### PR DESCRIPTION
**Pull request by jyoonbcit**
<details>
<summary>Release notes</summary>
<ul>
<li>Configured basic ESLint settings</li>
<li><s>Setup basic Git Hooks for pre-commit (EOF fixer, YAML check, trailing whitespace check)</s> (removed)</li>
<li>Setup pre-commit Prettier and ESLint checks</li>
<li>Create setup command to set up all local hooks into project</li>
</ul>
</details>

<p>
<s>After changing directories into dna-frontend, use <b>npm run prepare</b></div> on your local repo to setup the pre-commit checks (might want to use a virtual environment, this will install the pre-commit package).</s>

<b>UPDATE: </b>You no longer need to run npm run prepare; your initial <b>npm i</b> setup will run it for you.

In short, what this change will do is it will reformat any messy code and ensure that you cannot make commits that do not follow ESLint rules, and if you push messy code, then Prettier will reformat it for you so that all you have to do is stage it and commit it again.

### CODE REVIEW

**Review by m-a-s-h-e-d — COMMENTED** [review]
Approved.

Tested on `dna-frontend`, pre-commit hooks seem to be functioning properly. Developers still need to ensure that they manually run linting and formatting for files that are not staged for a commit.

Please wait for @FoRVaiS to check over `dna-backend` pre-commit hooks and confirm proper functionality before merging.

**Comment by FoRVaiS** [comment]
Hooks are functional for `dna-backend` as well.

Approved.

**Review by FoRVaiS — APPROVED** [review]
